### ISSUE TITLE
G2 - prevent large determinants

### ISSUE THREAD

**Issue by siwelwerd** [TeamBasedInquiryLearning/checkit-tbil-la#22]
Some of the determinants in the G2 problems are quite large/ugly
![G2](https://user-images.githubusercontent.com/30658947/117076467-0c1e9900-acfc-11eb-85b7-a86b9d2a923d.PNG)

**Comment by StevenClontz** [comment]
Yeah most of the public seeds are 100+. I wonder if reducing all numbers to [-4,4] would have a significant effect.

I'm working on the templating stuff but after that's done this seems like low-hanging fruit to pick off.

**Comment by siwelwerd** [comment]
I mean, the example there has |a_i,j|<=5, so I doubt it.  An easy fix might be to take the product of an upper triangular and a lower triangular (with small determinants, say <10).

**Comment by siwelwerd** [comment]
FWIW, Hadamard's inequality says that for a 4x4 matrix with |a_i,j|<=B, the determinant is (sharply) bounded by 16 B^4.

**Comment by StevenClontz** [comment]
> I mean, the example there has |a_i,j|<=5, so I doubt it. An easy fix might be to take the product of an upper triangular and a lower triangular (with small determinants, say <10).

Haven't toyed with this, but still imagine that you'll get a lot of 3-digit numbers, it's just that the final answer will be 236-218=18. Which I think is strictly better (I like puzzly problems where I can tell students to double-check their work if they get something out of a range), but doesn't really make the computations any nicer.

**Comment by siwelwerd** [comment]
So if you do LU, the first column will be a multiple of the last column of L (and the first row will be a multiple of the first row of U). And you can choose one of those to be nice. Then run some permutations to move the nice column/row around?

**Comment by StevenClontz** [comment]
Another idea: construct the 4x4 by starting with an invertible 2x2 with entries in [-5,5] and then work "backwards" through the path we'd recommend for students to solve.
